src/opm/io/eclipse/EclUtil.cpp

```diff
@@ -411,39 +461,50 @@ std::vector<T> Opm::EclIO::readBinaryArray(std::fstream& fileH, const int64_t si
 std::vector<int> Opm::EclIO::readBinaryInteArray(std::fstream &fileH, const int64_t size)
 {
     std::function<int(int)> f = Opm::EclIO::flipEndianInt;
-    return readBinaryArray<int,int>(fileH, size, Opm::EclIO::INTE, f);
+    return readBinaryArray<int,int>(fileH, size, Opm::EclIO::INTE, f, sizeOfInte);
 }
 
 
 std::vector<float> Opm::EclIO::readBinaryRealArray(std::fstream& fileH, const int64_t size)
 {
     std::function<float(float)> f = Opm::EclIO::flipEndianFloat;
-    return readBinaryArray<float,float>(fileH, size, Opm::EclIO::REAL, f);
+    return readBinaryArray<float,float>(fileH, size, Opm::EclIO::REAL, f, sizeOfReal);
 }
 
 
 std::vector<double> Opm::EclIO::readBinaryDoubArray(std::fstream& fileH, const int64_t size)
 {
     std::function<double(double)> f = Opm::EclIO::flipEndianDouble;
-    return readBinaryArray<double,double>(fileH, size, Opm::EclIO::DOUB, f);
+    return readBinaryArray<double,double>(fileH, size, Opm::EclIO::DOUB, f, sizeOfDoub);
 }
 
 std::vector<bool> Opm::EclIO::readBinaryLogiArray(std::fstream &fileH, const int64_t size)
 {
     std::function<bool(unsigned int)> f = [](unsigned int intVal)
                                           {
                                               bool value;
-                                              if (intVal == Opm::EclIO::true_value) {
```

Comment:
There are different schools - myself I have converted to the faith of "EarlyReturnIsOK" and would just return `true` or `false` instead of going through the temporary `value` variable.

Reply:
remove temporary variable value and using early return.